libraries/lib-ipc/IPCChannel.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <cstddef>
+
+/**
+ * \brief Interface for sending data from client to server or vise versa,
+ * complemented by IPCChannelStatusCallback
+ */
+class IPC_API IPCChannel
+{
+public:
+   virtual ~IPCChannel();
+   /**
+    * \brief Write data to the channel
+    * \param bytes Pointer to the data
+    * \param length Number of bytes to send
+    */
+   virtual void Send(const void* bytes, size_t length) = 0;
+};
+
+/**
+ * \brief Interface for listening connection status changes
+ */
+class IPC_API IPCChannelStatusCallback
+{
+public:
+   virtual ~IPCChannelStatusCallback();
+
+   /**
+    * \brief Called when connection attempt fails.
+    */
+   virtual void OnConnectionError() noexcept = 0;
+   /**
+    * \brief Called when connection established.
+    * \param channel Using this channel client or server can send data to the other side.
+    */
+   virtual void OnConnect(IPCChannel* channel) noexcept = 0;
```

Comment:
If I wrote this I would still prefer a reference argument to make clear it is never null

Reply:
Presence of reference itself does not guarantee nothing.

Reply:
It communicates clearly that channel is never null which is in fact true in usage but not clear just from reading this header.  You took my other suggestions to change a pointer to a reference.

Classes should have clearly expressed contracts in header files and one way to make things clear to those with a good knowledge of C++ is to use types in understood conventional ways.

Reply:
Not having this clearly stated, every override of OnConnect instead has the responsibility to null-check the argument.